src/olympia/versions/tests/test_models.py

```diff
@@ -969,6 +969,47 @@ def test_should_have_due_date_listed(self):
         )
         assert version.should_have_due_date
 
+    def test_should_have_due_date_listed_theme(self):
+        addon = Addon.objects.get(id=3615)
```

Comment:
I've seen different patterns in our tests, but wouldn't creating an addon in a factory make this test more atomic and therefore less flaky? Specifically:

- susceptible to test order failures
- susceptible to failure caused by changes in other tests

Reply:
Reading the tests I think I see another reason to use a factory pattern. 80% of the test is just setting up a version in the right state, this is done through modification of the model in the test. Is it not possible to create the model in the state you want to test against?

Couldn't you create a factory to create exactly that state and then you just pass a few args and assert against the returned model?

Reply:
Yes. We have such a factory - `addon_factory()`. We also have a bunch of old tests using fixed fixtures, and that infamous `3615` add-on is the most common of them. Typically when adding tests I try to copy the style of the tests nearby and that's why I went with re-using `3615` here, but I agree it's not ideal. I'll change it for the tests I've added here.

Reply:
Tests refactored to use `addon_factory()` in https://github.com/mozilla/addons-server/pull/22176/commits/b1add2b705c29223b0fbcf41806c32d184986dd6